content/blog/netlify-serverless-function-instagram-api/index.md

```diff
@@ -0,0 +1,151 @@
+
+### Step 2: Netlify Functions
+
+When you build and deploy sites on netlify, you can interact with third-party services and APIs with javascript places directly in your site. For example, you can have a script that sends event data to google analytics, or adds someone to a mailchimp list, or sends a request to a zapier webhook.
+
+But what if you want to write scripts with something other than javascript? Or if you want to do more complex things like resize images or query a database? Or if you have sensitive information in your script such as API tokens that you don’t want embedded on your site and visible to all? Or if there is no service or API that does what you need?
+
+That’s where [Netlify Functions](https://functions.netlify.com/) come in.
+
+Now that we have access to AWS Lambda Functions we can use one of the examples that you can find in Netlify site and create a Javascript script in your project like this:
+
+```
+const axios = require('axios')
+
+exports.handler = function instagram(event, context, callback) {
+  const endpoint = 'https://api.instagram.com/v1/users/self/media/recent'
+  const token = process.env.INSTAGRAM_ACCESS_TOKEN
+  const limit = 5
+
+  axios
+    .get(`${endpoint}?access_token=${token}&count=${limit}`)
+    .then(({ data: { data: posts } }) => {
+      callback(null, {
+        statusCode: 200,
+        headers: {
+          'content-type': 'application/json',
+        },
+        body: JSON.stringify(
+          posts.map(i => ({
+            id: i.id,
+            link: i.link,
+            images: i.images,
+            videos: i.videos,
+            caption: i.caption.text,
+          })),
+        ),
+      })
+    })
+    .catch((e) => {
+      callback(e)
+    })
+}
+```
+
+### Step 3: Testing your Lambda
+
+As you can see, at this point we need to install some node libraries like Axios in order to fetch data from Instagram API. Also, you can test your code locally using netlify-lambda.
+
+`npm i netlify-lambda axios`
+
+With that, you can add a command into your package.json in order to serve your lambda function using netlify-lambda, you have to specify the location of your lambda function. In my case I created a `./functions` folder and added my lambda functions there:
+
+```
+"scripts": {
+    "lambda-serve": "netlify-lambda serve functions",
+    "lambda-build": "netlify-lambda build functions"
+  },
+```
+
+Another useful step is create a `netlify.toml` file in the root of your project and you can add the some commands like this:
+
+```
+[build]
+    functions = "lambda"
+    Command = "npm run lambda-build"
+```
+
+Have in mind that you can also [specify the location of your lambda functions](https://docs.netlify.com/functions/configure-and-deploy/#configure-the-functions-folder) in netlify settings for your project, but if you use **netlify.toml** file it will override your configuration.
```

Comment:
```suggestion
Keep in mind that you can also [specify the location of your lambda functions](https://docs.netlify.com/functions/configure-and-deploy/#configure-the-functions-folder) in netlify settings for your project, but if you use **netlify.toml** file it will override your configuration.
```